tests/sentinel/tests/03-runtime-reconf.tcl

```diff
@@ -1 +1,156 @@
+    assert_equal {OK} [R $id CONFIG SET masterauth ""]
+}
+
+proc verify_replicas_flags {id} {
+    foreach replica [S $id SENTINEL REPLICAS mymaster] {
+        if {[string match "*down*" [dict get $replica flags]]} {
+            return 0
+        }
+    }
+    return 1
+}
+
+proc check_instances_status {} {
+    foreach_sentinel_id id {
+        wait_for_condition 1000 50 {
+            ![string match "*down*" [dict get [S $id SENTINEL MASTER mymaster] flags]]
+        } else {
+            fail "At least some sentinel can't connect to master"
+        }
+    }
+    foreach_sentinel_id id {
+        wait_for_condition 1000 50 {
+            [verify_replicas_flags $id]
+        } else {
+            fail "At least some sentinel can't connect to replica"
+        }
+    }
+}
+
+proc re_monitor_mymaster {} {
+    foreach_sentinel_id id {
+        catch {S $id SENTINEL REMOVE mymaster}
+    }
+
+    set master_id 0
+    set sentinels [llength $::sentinel_instances]
+    # avoid failover
+    set quorum [expr {$sentinels + 1}]
+    foreach_sentinel_id id {
+        S $id SENTINEL MONITOR mymaster \
+              [get_instance_attrib redis $master_id host] \
+              [get_instance_attrib redis $master_id port] $quorum
+    }
+}
+
+test "(server-auth-init) Set up requirepass configuration" {
+    setup_server_auth
+}
+
+test "All servers have non-empty requirepass before monitored" {
+    re_monitor_mymaster
+    foreach_sentinel_id id {
+        assert {[S $id sentinel master mymaster] ne {}}
+        assert_equal {OK} [S $id SENTINEL SET mymaster down-after-milliseconds 18000]
+        assert_equal {OK} [S $id SENTINEL SET mymaster auth-pass $::password]
+    }
+
+    after 20000
+    check_instances_status
+}
+
+test "Only replicas have non-empty requirepass before monitored" {
+    assert_equal {OK} [R $master_id CONFIG SET requirepass ""]
+    re_monitor_mymaster
+    assert_equal {OK} [R $master_id CONFIG SET requirepass $::password]
+    foreach_sentinel_id id {
+        assert {[S $id sentinel master mymaster] ne {}}
+        assert_equal {OK} [S $id SENTINEL SET mymaster down-after-milliseconds 18000]
+        assert_equal {OK} [S $id SENTINEL SET mymaster auth-pass $::password]
+    }
+
+    after 20000
+    check_instances_status
+}
+
+test "(server-auth-cleanup) Tear down requirepass configuration" {
+    teardown_server_auth
+}
+
+test "(server-ACL-init) Set up ACL configuration" {
+    foreach_redis_id id {
+        setup_server_acl $id
+    }
+    assert_equal $::user [R 0 ACL WHOAMI]
+}
+
+test "All servers have ACL configuration before monitored" {
+    re_monitor_mymaster
+    foreach_sentinel_id id {
+        assert {[S $id sentinel master mymaster] ne {}}
+        assert_equal {OK} [S $id SENTINEL SET mymaster down-after-milliseconds 18000]
+        assert_equal {OK} [S $id SENTINEL SET mymaster auth-user $::user]
+        assert_equal {OK} [S $id SENTINEL SET mymaster auth-pass $::password]
+    }
+
+    after 20000
```

Comment:
Is there a particular reason that such a long delay is needed? Can the test be optimized instead to have a much shorter `down-after-milliseconds` and a smaller delay here, or even use `wait_for_condition` already in this level?

Sentinel tests are already very slow and we should avoid making them slower if it's not absolutely necessary.

Reply:
When the value of `down-after-milliseconds` is greater than `SENTINEL_MIN_LINK_RECONNECT_PERIOD`, the sentinel with a version lower than 5.0 can pass "auth test".

Reply:
@yz1509 @hwware The problem is sentinel tests are already very slow, and we shouldn't slow them much further. Do you think it makes sense to make this define configurable, at least for testing purposes?

Reply:
@yossigo I agree with you.

Reply:
@yossigo Yes, I totally agree with you, and I will create a separated PR for this configuration.